src/index/interrupt.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "knowhere/index/interrupt.h"
+
+#include "folly/futures/Future.h"
+namespace knowhere {
+#ifdef KNOWHERE_WITH_CARDINAL
+Interrupt::Interrupt(const std::chrono::seconds& timeout) : start(std::chrono::steady_clock::now()), timeout(timeout) {
+}
+#else
+Interrupt::Interrupt() = default;
+#endif
+
+#ifdef KNOWHERE_WITH_CARDINAL
+void
+Interrupt::Stop() {
```

Comment:
Why we switch the name from `Handle` to `Interrupt`? `Handler` sounds more precise to me, since it is responsible for interrupting, status check, waiting, etc.

Reply:
@liliu-z Alex suggested in the corresponding PR for Cardinal that the name should be more specific, as "handle" is too generic.